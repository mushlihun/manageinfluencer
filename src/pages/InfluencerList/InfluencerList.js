import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import FavouriteButton from '../../components/FavouriteButton/FavouriteButton.js'
import Button from '../../components/Button/Button.js'

export default function InfluencerList({ influencerData, onFavouriteClick }) {
  return (
    <WrapperStyled className="profiles">
      {influencerData?.map((profile) => {
        return (
          <section key={profile.id}>
            <ContainerStyled>
              <ImageStyled src={profile.image} />
              <UsernameStyled>{profile.username}</UsernameStyled>
              <CategoriesStyled>
                {profile.categories.join(', ')}
              </CategoriesStyled>
              <FavouriteButton
                onClick={() => onFavouriteClick(profile)}
                isFavourite={profile.isFavourite}
              />
            </ContainerStyled>
            <DetailListStyled>
              <DetailItemStyled key={profile.username}>
                Follower: {profile.follower.toLocaleString('en-US')}
              </DetailItemStyled>
              <DetailItemStyled key={profile.gender}>
                Gender: {profile.gender}
              </DetailItemStyled>
              <DetailItemStyled key={profile.age}>
                Age: {profile.age}
              </DetailItemStyled>
              <DetailItemStyled key={profile.location}>
                Location: {profile.location}
              </DetailItemStyled>
              <DetailItemStyled key={profile.languages}>
                Languages: {profile.languages.join(', ')}
              </DetailItemStyled>
            </DetailListStyled>
            <AnchorStyled href="mailto:influencer@gmail.com">
              <Button id="contact-button" headline="Contact" />
            </AnchorStyled>
          </section>
        )
      })}
    </WrapperStyled>
  )
}

const WrapperStyled = styled.section`
  margin: 0px auto;
  max-width: 400px;
`

const ContainerStyled = styled.section`
  background: var(--transparent-blue);
  height: 300px;
`

const ImageStyled = styled.img`
  width: 120px;
  height: 190px;
  display: block;
  margin-left: auto;
  margin-right: auto;
  padding: 20px 0 0 0;
  border-radius: 50%;
`

const UsernameStyled = styled.h2`
  font-family: 'Poppins';
  font-weight: var(--font-weight-semibold);
  font-size: 20px;
  text-align: center;
  padding: 10px 0 0 0;
  margin-top: 0;
  margin-bottom: 0;
`
const CategoriesStyled = styled.h3`
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  font-size: 18px;
  text-align: center;
  margin-top: 0;
  margin-bottom: 0;
`

const DetailListStyled = styled.ul`
  text-align: left;
  margin: 0;
  list-style-type: none;
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  margin: 50px 0 40px 0;
  width: 300px;
  margin-left: auto;
  margin-right: auto;
`
const DetailItemStyled = styled.li`
  letter-spacing: 1px;
  margin: 10px 0;
  position: relative;
`
const AnchorStyled = styled.a`
  text-decoration: none;
`

InfluencerList.propTypes = {
  influencerData: PropTypes.array,
}
