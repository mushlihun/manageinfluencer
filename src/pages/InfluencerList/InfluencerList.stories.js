import React from 'react'
import InfluencerList from './InfluencerList'
import influencerData from '../../mocks/influencer.json'
import HeaderInfluencerList from '../../components/Header/HeaderInfluencerList'

export default {
  title: 'Pages/InfluencerList',
  component: InfluencerList,
}

export const fullList = () => {
  return (
    <>
      <HeaderInfluencerList headline="Browse" />
      <InfluencerList influencerData={influencerData} />
    </>
  )
}
