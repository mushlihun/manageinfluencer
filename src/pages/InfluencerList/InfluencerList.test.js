import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import InfluencerList from './InfluencerList.js'
import 'jest-styled-components'

describe('InfluencerList', () => {
  const influencerData = [
    {
      id: 1,
      username: '@fitness4life',
      categories: ['Fitness'],
      follower: '10k',
      gender: 'female',
      age: '25 years',
      location: 'Hamburg',
      languages: ['German, ', 'English'],
    },
    {
      id: 2,
      username: '@VeganPlace',
      categories: ['Food', 'Health'],
      follower: '1.000',
      gender: 'female',
      age: '28 years',
      location: 'Munich',
      languages: ['German, ', 'English'],
    },
  ]

  it('displays the information details of an influencer', () => {
    const { getByText, getAllByText } = render(
      <InfluencerList influencerData={influencerData} />
    )
    influencerData.forEach(
      ({
        username,
        follower,
        location,
        age,
        gender,
        languages,
        categories,
      }) => {
        expect(getByText(username)).toBeInTheDocument()
        expect(getByText(new RegExp(follower, 'i'))).toBeInTheDocument()
        expect(getByText(new RegExp(age, 'i'))).toBeInTheDocument()
        expect(getByText(new RegExp(location, 'i'))).toBeInTheDocument()
        expect(getAllByText(new RegExp(gender, 'i'))[0]).toBeInTheDocument()
        expect(JSON.stringify(new RegExp(languages, 'i'))).toBe(
          JSON.stringify(/languages/i)
        )
        expect(JSON.stringify(new RegExp(categories, 'i'))).toBe(
          JSON.stringify(/categories/i)
        )
      }
    )
  })

  it('renders influencer details correctly', () => {
    const tree = renderer.create(
      <InfluencerList influencerData={influencerData} />
    )
    expect(tree).toMatchSnapshot()
  })
})
