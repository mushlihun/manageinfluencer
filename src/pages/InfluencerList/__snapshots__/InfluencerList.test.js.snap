// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfluencerList renders influencer details correctly 1`] = `
.c5 {
  background: var(--lightpink);
  border-radius: 50%;
  text-align: center;
  height: 60px;
  width: 60px;
  display: block;
  margin: auto;
  margin-top: 15px;
  border: none;
  stroke: #e693b5;
}

.c6 {
  margin-top: 3px;
}

.c10 {
  color: var(--main-font-color);
  font-family: var(--font-weight-light);
  background: var(--darkgreen);
  border: var(--darkgreen);
  box-shadow: 0 3px 10px var(--boxshadow-color);
  font-size: 18px;
  border-radius: 40px;
  width: 130px;
  height: 75px;
  display: block;
  border: none;
  margin-right: auto;
  margin-left: auto;
  margin-bottom: 60px;
  cursor: pointer;
}

.c0 {
  margin: 0px auto;
  max-width: 400px;
}

.c1 {
  background: var(--transparent-blue);
  height: 300px;
}

.c2 {
  width: 120px;
  height: 190px;
  display: block;
  margin-left: auto;
  margin-right: auto;
  padding: 20px 0 0 0;
  border-radius: 50%;
}

.c3 {
  font-family: 'Poppins';
  font-weight: var(--font-weight-semibold);
  font-size: 20px;
  text-align: center;
  padding: 10px 0 0 0;
  margin-top: 0;
  margin-bottom: 0;
}

.c4 {
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  font-size: 18px;
  text-align: center;
  margin-top: 0;
  margin-bottom: 0;
}

.c7 {
  text-align: left;
  margin: 0;
  list-style-type: none;
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  margin: 50px 0 40px 0;
  width: 300px;
  margin-left: auto;
  margin-right: auto;
}

.c8 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin: 10px 0;
  position: relative;
}

.c9 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<section
  className="c0 profiles"
>
  <section>
    <section
      className="c1"
    >
      <img
        className="c2"
      />
      <h2
        className="c3"
      >
        @fitness4life
      </h2>
      <h3
        className="c4"
      >
        Fitness
      </h3>
      <section>
        <button
          className="c5"
          id="like-button"
          onClick={[Function]}
        >
          <svg
            className="c6"
            fill="var(--lightpink)"
          >
            heart.svg
          </svg>
        </button>
      </section>
    </section>
    <ul
      className="c7"
    >
      <li
        className="c8"
      >
        Follower: 
        10k
      </li>
      <li
        className="c8"
      >
        Gender: 
        female
      </li>
      <li
        className="c8"
      >
        Age: 
        25 years
      </li>
      <li
        className="c8"
      >
        Location: 
        Hamburg
      </li>
      <li
        className="c8"
      >
        Languages: 
        German, , English
      </li>
    </ul>
    <a
      className="c9"
      href="mailto:influencer@gmail.com"
    >
      <button
        className="c10"
      >
        Contact
      </button>
    </a>
  </section>
  <section>
    <section
      className="c1"
    >
      <img
        className="c2"
      />
      <h2
        className="c3"
      >
        @VeganPlace
      </h2>
      <h3
        className="c4"
      >
        Food, Health
      </h3>
      <section>
        <button
          className="c5"
          id="like-button"
          onClick={[Function]}
        >
          <svg
            className="c6"
            fill="var(--lightpink)"
          >
            heart.svg
          </svg>
        </button>
      </section>
    </section>
    <ul
      className="c7"
    >
      <li
        className="c8"
      >
        Follower: 
        1.000
      </li>
      <li
        className="c8"
      >
        Gender: 
        female
      </li>
      <li
        className="c8"
      >
        Age: 
        28 years
      </li>
      <li
        className="c8"
      >
        Location: 
        Munich
      </li>
      <li
        className="c8"
      >
        Languages: 
        German, , English
      </li>
    </ul>
    <a
      className="c9"
      href="mailto:influencer@gmail.com"
    >
      <button
        className="c10"
      >
        Contact
      </button>
    </a>
  </section>
</section>
`;
